backend/blueprints/spa_api/service_layers/replay/replay_positions.py

```diff
@@ -0,0 +1,79 @@
+import ast
+import gzip
+import os
+
+import pandas as pd
+
+from typing import List, cast
+from flask import current_app
+
+from carball.analysis.utils import proto_manager, pandas_manager
+from backend.database.objects import Game, PlayerGame
+from .replay_player import ReplayPlayer
+from ..utils import sort_player_games_by_team_then_id
+from ...errors.errors import ReplayNotFound, ErrorOpeningGame
+
+
+class ReplayPositions:
+    def __init__(self, id_: str,
+                 ball, players, colors, names,
+                 frames):
+        self.id = id_
+
+        self.ball = ball
+        self.players = players
+        self.colors = colors
+        self.names = names
+        self.frames = frames
+
+    @staticmethod
+    def create_from_id(id_: str) -> 'ReplayPositions':
+        pickle_path = os.path.join(current_app.config['PARSED_DIR'], id_ + '.replay.pts')
+        gzip_path = os.path.join(current_app.config['PARSED_DIR'], id_ + '.replay.gzip')
+        replay_path = os.path.join(current_app.config['REPLAY_DIR'], id_ + '.replay')
+        if os.path.isfile(replay_path) and not os.path.isfile(pickle_path):
+            raise ReplayNotFound()
+
+        try:
+            with gzip.open(gzip_path, 'rb') as f:
+                data_frame = pandas_manager.PandasManager.safe_read_pandas_to_memory(f)
```

Comment:
We should do timing on this.  I'm wondering if it would be better to send like the first 2s and only load the rest of the press play.

Reply:
Definitely, this endpoint is sending a lot of data

Reply:
Yeah that's why I wonder about timing.

So what I'm thinking is that when we have the replay viewer become visible it has 2s in it or starts requesting the first 2s.

Then when the user presses play we send the rest of the data.

But if that ends up being the same in server load / sending them it doesn't matter

Reply:
So it is more on bandwidth server load than pure CPU load.